sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ImplementationBridgeHelpers.java

```diff
@@ -1124,4 +1127,43 @@ CosmosClientTelemetryConfig createSnapshot(
                 boolean effectiveIsClientTelemetryEnabled);
         }
     }
+
+    public static final class CosmosContainerIdentityHelper {
+
+        private static final AtomicReference<Boolean> cosmosContainerIdentityClassLoaded = new AtomicReference<>(false);
+        private static final AtomicReference<CosmosContainerIdentityAccessor> accessor = new AtomicReference<>();
+
+        private CosmosContainerIdentityHelper() {}
+
+        public static CosmosContainerIdentityAccessor getCosmosContainerIdentityAccessor() {
+
+            if (!cosmosContainerIdentityClassLoaded.get()) {
+                logger.debug("Initializing CosmosContainerIdentityAccessor...");
+                initializeAllAccessors();
+            }
+
+            CosmosContainerIdentityAccessor snapshot = accessor.get();
+
+            if (snapshot == null) {
+                logger.error("CosmosContainerIdentityAccessor is not initialized yet!");
+                System.exit(9924); // Using a unique status code here to help debug the issue.
+            }
+
+            return snapshot;
+        }
+
+        public static void setCosmosContainerIdentityAccessor(final CosmosContainerIdentityAccessor newAccessor) {
+            if (!accessor.compareAndSet(null, newAccessor)) {
```

Comment:
What if newAccessor is null? Should there be a check for that?

Reply:
Fixed.